include/zephyr/drivers/zephyr_thermistor.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_THERMISTOR_H
+#define ZEPHYR_THERMISTOR_H
+
+#include <zephyr/types.h>
+
+struct ntc_compensation {
+	const int32_t temp_c;
+	const uint32_t ohm;
+};
```

Comment:
hey does this need to live in `include/zephyr/drivers`? It's only used within the driver, if it should not be exposed to the driver users it should probably be declared inside the driver itself (that is `zephyr_thermistor.c`). You may want to add an extra local header files if you need to share some stuff between the various thermistor files but this should only have the structs part of the "public" API.

Reply:
Good catch!